services/thumbnails/README.md

```diff
@@ -0,0 +1,29 @@
+# Thumbnails service
+
+The thumbnails service provides methods to generate thumbnails for various files.
+It retrieves the sources at the location where the user files are stored and saves the tumbnails at the location where system files are stored. Those locations have defaults but can be manually defined via environment variables. The relevant environment variables are:
+
+-  `OCIS_BASE_DATA_PATH` which will contain system relevant data and
+-  `STORAGE_USERS_OCIS_ROOT` which will contain the user source data.
+-  `THUMBNAILS_FILESYSTEMSTORAGE_ROOT` used if thumnails should be separated from system files.
+
+For details and defaults see the documentation.
+
+Thumbnails can be generated for the following file types:
+
+-  png
+-  jpg
+-  gif
+-  tiff
+-  bmp
+-  txt
+
+Thumbnails can either be generated as `png`, `jpg` or `gif` files, various resolutions can be defined via `THUMBNAILS_RESOLUTIONS`.
```

Comment:
> The generated image types can be requested by the caller via a parameter.

* Which means that always both image types are generated which contradicts the generate part of the sentence above - or?
* How is that done that a thumb can be requested as a specific type?

Reply:
See my answer here: https://github.com/owncloud/ocis/pull/4737#discussion_r987044139

Reply:
ok, I read it, I have to refine the question,

> The caller can specify the thumbnail type

1. Who is the "caller" and how does he know about the image types available?
2. The available image types are hardcoded in the thumbnail service, no env availabe - or?
3. The request can contain a resolution (W x H), we have an env defining the available resolutions. How does the caller know the available resolutions to select one available for the request?

`message GetThumbnailRequest` has a well defined content for the thumbnail service to generate the requested thumbnail. In a nutshell, how does the caller know about the options to define? and how does the caller identify the set to request?

Reply:
> Who is the "caller" and how does he know about the thumb types available?

It's another service or later it could be a client like the Desktop client or an App.
The available types are defined in the protobuf service definition

> Just to be sure, the available thumb types are hardcoded in the thumbnail service as described in the example abobe, no env availabe - or?

Yes, they are defined in the service definition and currently not configurable.

> The request can contain a resolution (W x H), we have an env defining the available resolutions. How does the caller know the available resolutions to select one available for the request?

The caller doesn't. It can request any arbitrary resolution and the thumbnail service will use the one closest to the requested resolution.
E.g.:

Requested: 18x12
Available: 30x20, 15x10, 9x6
Returned: 15x10

Reply:
I will add some text about the resolutions.
But I am still not clear about what the circumstances are, that a service choses one type over another - out of the available. I got it that the service knows because of the definition, it now can select one out of the available, but how is the criteria defined to select one?

Reply:
> I got it that the service knows the available because of the definition, it now can select one, but what is the criteria to select one?

That is up to the developer of the calling service. The service could always request `png` files or it could always request `jpg` files because of smaller file sizes or it can make it dependent on some other criteria.